mlir/lib/Dialect/Arith/IR/ArithDialect.cpp

```diff
@@ -46,6 +49,17 @@ void arith::ArithDialect::initialize() {
       >();
   addInterfaces<ArithInlinerInterface>();
   declarePromisedInterface<ArithDialect, ConvertToLLVMPatternInterface>();
+  declarePromisedInterface<SelectOp,
+                           bufferization::BufferDeallocationOpInterface>();
+  declarePromisedInterface<ConstantOp,
```

Comment:
How does this work? `MLIRArithDialect` has no CMake dependency on `MLIRBufferizationDialect` as far as I know.

Reply:
Cmake doesn't need a dependency to use a header, and it looks like everything needed here is contained in headers because templates.

Whether we _conceptually_ want a dependency here is a good question.